periodic/periodic.go

```diff
@@ -488,6 +489,32 @@ func (r *periodicRunner) Run() RunnerResults {
 	return result
 }
 
+type AccessLogger struct {
+	mu     sync.Mutex
+	file   *os.File
+	format string
+}
+
+func NewAccessLogger(file, format string) (*AccessLogger, error) {
+	f, err := os.OpenFile(file, os.O_APPEND|os.O_CREATE|os.O_WRONLY, 0644)
+	if err != nil {
+		return nil, err
+	}
+	return &AccessLogger{file: f, format: format}, nil
+}
+
+func (a *AccessLogger) Report(thread int, time int64, latency float64) {
+	a.mu.Lock()
+	switch a.format {
+	case "influx":
+		// https://docs.influxdata.com/influxdb/v2.1/reference/syntax/line-protocol/
+		fmt.Fprintf(a.file, `latency,thread=%d value=%f %d`+"\n", thread, latency, time)
```

Comment:
is the compiler smart enough to not keep redoing the \n concat? if not make that constant once outside or for this case just one string as there is no " to \"